test/EndToEndSpec.hs

```diff
@@ -958,6 +992,21 @@ spec = around_ (inTempDirectoryNamed "my-package") $ do
                 cxx-options: -Wall
           |]) {packageCabalVersion = "2.2"}
 
+    describe "asm-sources" $ before_ (touch "foo.asm" >> touch "asmbits/bar.asm") $ do
+      it "accepts asm-sources" $ do
+        [i|
+        executable:
+          asm-sources:
+            - foo.asm
+            - asmbits/*.asm
+        |] `shouldRenderTo` (executable_ "my-package" [i|
+        autogen-modules:
+            Paths_my_package
+        asm-sources:
+            foo.asm
+            asmbits/bar.asm
+        |]) {packageCabalVersion = "3.0"}
+
```

Comment:
Maybe move this up, just below the test case for `asm-options`?

Reply:
Will do.